src/image_handlers/imagehandler.h

```diff
@@ -20,27 +20,32 @@
 
 #include <memory>
 #include <QList>
-#include<algorithm>
+#include <algorithm>
 #include "commons/frame.h"
 
 class ImageHandler {
+protected:
+    virtual void doHandle(Frame::ptr frame) = 0;
+
 public:
   typedef std::shared_ptr<ImageHandler> ptr;
-  virtual void handle(const Frame::ptr &frame) = 0;
```

Comment:
Originally I was planning on applying the user-selected endianess here. Still, I suggest to keep this split into stable public interface/virtual implementation, might come handy in the future.